draft-ietf-httpbis-compression-dictionary.md

```diff
@@ -264,6 +264,37 @@ longest "match-query" takes precedence.
 1. Given equivalent destination, path and search precedence, the most recently
 fetched dictionary takes precedence.
 
+## Content-Dictionary
+
+When a HTTP server responds with a resource that is encoded with a dictionary
+the server SHOULD send the hash of the dictionary that was used in the
+"Content-Dictionary" response header.
+
+The "Content-Dictionary" response header is a lowercase Base16-encoded
+{{RFC4648}} {{SHA-256}} hash of the contents of the dictionary that was used
+to encode the response.
+
+If the HTTP response contains a "Content-Dictionary" response header with the
+hash of a dictionary that the client does not have available then the client
+MUST NOT attempt to decode or use the HTTP response.
```

Comment:
A "MUST NOT" here doesn't really help.  "cannot" is probably the word you want.

Reply:
Is "cannot" the right way to phrase it when it is technically possible but bad things could happen? i.e. You can decompress a resource using a dictionary other than the one that was used to compress it but you will get the wrong output (level of brokenness depends on how much the dictionaries varied).  So it's "possible" to do but you absolutely must not do it or you'll get corrupt data.

Reply:
"cannot decode or use" would probably be correct. The client could _attempt_ to decode, but the decoding will not be successful, and a content-digest would reveal that.

Reply:
Thanks.  Changed to cannot.  I was just worried that it was a technical barrier that people might try to rely on (decode and trust that it would fail if the dictionary didn't match) instead of failing silently by generating garbage.